libcaf_core/caf/detail/parser/read_uri.hpp

```diff
@@ -116,6 +117,19 @@ void read_uri(State& ps, Consumer&& consumer) {
   auto set_path = [&] { consumer.path(take_str()); };
   auto set_host = [&] { consumer.host(take_str()); };
   auto set_userinfo = [&] { consumer.userinfo(take_str()); };
+  auto set_colon_position = [&] {
+    colon_position = str.size();
+    str.push_back(':');
+  };
+  auto set_host_and_port = [&]() -> bool {
```

Comment:
I didn't try that version, thanks for the idea. I'll give it a shot.

Reply:
The implementation of the parser will just `goto fsm_fin;` where it would be set to `pec::success`. I've had to add a line for error checking. Also, if I don't set it to `pec::success` prior calling the exit statement, it would break. I guess this isn't that scary since the default constructed state also holds a `pec::success`.